src/js/bookmarksPaginator.js

```diff
@@ -0,0 +1,295 @@
+/*jslint node: true */
+'use strict';
+let _ = require('lodash');
+let constants = require("./constants");
+let common = require('./common');
+
+/**
+ * Wrapper around ordinary request. Sends request with prefilled access token.
+ * @param url String API endpoint
+ * @param data JSON
+ * @returns {Promise}
+ */
+function request(url, data) {
+  return new Promise(function(resolve,reject){
+    chrome.runtime.sendMessage(null, {
+      command: 'sendAuthorizedPostRequest',
+      url: url,
+      data: data
+    }, function(response) {
+      if (response.error) {
+        reject(response.error);
+      } else if (chrome.runtime.lastError) {
+        reject('Request didn\'t return valid result');
+      } else {
+        resolve(response);
+      }
+    });
+  });
+}
+
+// util and helper functions. All of them static
+var moment = require("moment");
+class BookmarksTransformer {
+  static getBookmarksFromBookmarksList(bookmarksList) {
+    return Object.keys(bookmarksList).map(bookmarkId => bookmarksList[bookmarkId]);
+  }
+  static sortBookmarksByNewest(bookmarks) {
+    return _.chain(bookmarks)
+      // Sort by name and time.updated. It's a stable sort so we sort it two
+      // times to make sure the items won't jump around as they're sorted
+      // (which could happen if we were sorting just by time which is not
+      // unique).
+      .sortBy('item_id')
+      .sortBy('time_updated')
+      .reverse()
+      .value();
+  }
+
+  static parseDomain(url) {
+    return url.match(/^((http[s]?|ftp):\/)?\/?([^:\/\s]+)(:([^\/]*))?/i)[3] || '';
+  }
+  // definitelly going to be refactored, I just want to get the release working ASAP
+  static processItem(item) {
+    // Real URL is preferably the resolved URL but could be the given URL
+    let realURL = item.resolved_url || item.given_url;
+    item.url = realURL;
+
+    // Regular expression to parse out the domain name of the URL, or an empty string if something fails
+    item.domain = BookmarksTransformer.parseDomain(item.url);
+    // Fetches a icon from a great webservice which provides a default fallback icon
+    item.icon = 'https://web-image.appspot.com/?url=' + realURL;
+
+    item.tags = _.isObject(item.tags) ? Object.keys(item.tags) : [];
+
+    item.title = item.resolved_title || item.given_title;
+
+    item.time = {
+      added: moment.unix(item.time_added),
+      updated: moment.unix(item.time_updated),
+      read: moment.unix(item.time_read),
+      favorited: moment.unix(item.time_favorited)
+    };
+
+    item.favorite = parseInt(item.favorite) === 1;
+    item.status = parseInt(item.status);
+
+    return item;
+  }
+}
+
+// their main responsibility is to handle pagination
+class BookmarksPaginatorInterface {
```

Comment:
not sure if this was a good idea, but I wanted to play with ES6 classes. However it ended up being a mess, so I completely reduced all the inheritance and use it only as an 'interface'.